knime_extension/src/nodes/locationanalysis.py

```diff
@@ -0,0 +1,621 @@
+        problem += pulp.lpSum(object_list)
+
+        for i in range(num_trt):
+            constrain = []
+            for j in range(num_hosp):
+                within = df[(df[self.DemandID] == i) & (df[self.SupplyID] == j)][
+                    "within"
+                ].values[0]
+                constrain.append(within * varX["X" + " " + str(j)])
+        problem += pulp.lpSum(constrain) >= 1
+
+        problem.solve()
+        # print(pulp.LpStatus[problem.status])
+
+        DemandPt["assignSID"] = None
+        DemandPt["SIDcoord"] = None
+
+        area = pd.DataFrame(columns=["OID"])
+        for it in X:
+            v = varX[it]
+            if v.varValue == 1:
+                print(1)
+                j = getIndex(it)
+                tempdict = {"OID": j}
+                area = area.append(tempdict, ignore_index=True)
+
+        for it in range(num_trt):
+            index = pd.to_numeric(
+                df[(df[self.DemandID] == it) & (df[self.SupplyID].isin(area["OID"]))][
+                    self.ODcost
+                ]
+            ).idxmin()
+            oid = int(df.at[index, self.SupplyID])
+            DemandPt.at[it, "assignSID"] = SupplyPt.at[oid, self.SupplyID]
+            DemandPt.at[it, "SIDcoord"] = SupplyPt.at[oid, "geometry"]
+        DemandPt["linxy"] = [
+            LineString(xy) for xy in zip(DemandPt["geometry"], DemandPt["SIDcoord"])
+        ]
+        SIDwkt = gp.GeoDataFrame(geometry=DemandPt.SIDcoord, crs=df.crs)
+        Linewkt = gp.GeoDataFrame(geometry=DemandPt.linxy, crs=df.crs)
+        DemandPt["SIDwkt"] = SIDwkt.geometry
+        DemandPt["Linewkt"] = Linewkt.geometry
+        DemandPt = DemandPt.drop(columns=["linxy", "SIDcoord"])
+        DemandPt = DemandPt.reset_index(drop=True)
+        return knext.Table.from_pandas(DemandPt)
+
+
+############################################
+# Location-allocation MCLP
+############################################
+@knext.node(
+    name="MCLP",
+    node_type=knext.NodeType.MANIPULATOR,
+    icon_path=__NODE_ICON_PATH + "MCLP.png",
+    category=__category,
+    after="",
+)
+@knext.input_table(
+    name="Input OD list with geometries ",
+    description="Table with geometry information of demand and supply",
+)
+@knext.output_table(
+    name="Demand table with MCLP result",
+    description="Demand table with assigned supply point and link",
+)
+@knut.pulp_node_description(
+    short_description="Solve MCLP problem to Maximize Capacitated Coverage by setting an impedance cutoff.",
+    description="The MCLP model, maximum covering location problem (MCLP), aims to Locate p facilities,"
+    + " and demand is covered if it is within a specified distance (time) of a facility. "
+    + "The P-Median problem will be solved by PuLP package. ",
+    references={
+        "Pulp.Solver": "https://coin-or.github.io/pulp/guides/how_to_configure_solvers.html",
+    },
+)
+class MCLPNode:
+
+    DemandID = knext.ColumnParameter(
```

Comment:
the variable names in this file follow `PascalCase` notation while in Python we usually use `snake_case`